en/guide/define_xml_element.md

```diff
@@ -0,0 +1,424 @@
+# How to Define MAVLink Messages & Enums
+
+MAVLink enums, messages, commands, and other elements are [defined within XML files](../messages/README.md) and then converted to libraries for [supported programming languages](../README.md#supported_languages) using a *code generator*.
+
+This topic provides practical guidance for defining and extending MAVLink XML elements, including conventions and best-practice.
+
+> **Note** For detailed information about the file format see [MAVLink XML Schema](../guide/xml_schema.md#messages) (you can also inspect [common.xml](https://github.com/mavlink/mavlink/tree/master/message_definitions/v1.0/common.xml) and other dialect files). 
+
+## Should we Define Messages or Commands?
```

Comment:
FYI @amilcarlucas , new info about commands vs messages. I added to top here as it is possibly first decision someone should consider.